src/PaymentPart/Output/AbstractOutput.php

```diff
@@ -48,6 +54,26 @@ public function isPrintable(): bool
         return $this->printable;
     }
 
+    public function setQrCodeImageFormat(string $imageFormat): self
+    {
+        if (!in_array($imageFormat, QrCode::SUPPORTED_FILE_FORMATS)) {
+            throw new UnsupportedFileExtensionException(sprintf(
+                'The qr code in your payment part cannot be created. Only these file extensions are supported: %s. You provided: %s.',
+                implode(', ', QrCode::SUPPORTED_FILE_FORMATS),
+                $imageFormat
+            ));
```

Comment:
I am not yet happy with this part. This is duplicated code from the `QrCode` class and also uses its data. Is there a better way to do it?